### DISCUSSION TITLE
how to use Filter?

### DISCUSSION THREAD

**Discussion by 0x0bit** [mikro-orm/mikro-orm#5870] [Q&A] [answered]
<img width="469" alt="image" src="https://github.com/user-attachments/assets/47fca2ce-d660-409f-9a62-0c718c70c77b">
<img width="643" alt="image" src="https://github.com/user-attachments/assets/2601eef8-7260-4b68-b89a-40cbf254406e">

`createQueryBuilder` how to use Filter?

**Comment by B4nan** [comment, marked as answer]
You can use `em.applyFilters()` to create the filter query and use that via `qb.where()`.

```ts
const qb = ...;
const filters = await em.applyFilters(Type, {}, {}, 'read');
qb.andWhere(filters);
...
```

Also next time please don't open issues like this, if you need help, ask on discord or in GH discussions, issues are for actual bugs and feature requests, not for support.

**Reply by 0x0bit** [reply]
I'm sorry, thank you for your answer, it was very helpful.

**Reply by B4nan** [reply]
Yeah, no worries. Btw I might add some helper to the QB interface to do this for you, maybe even into the next version, so check the changelogs for this. It has been a common issue so having something like `qb.applyFilters()` could make a lot of sense (and would be a simple wrapper over the code I posted above).

**Reply by 0x0bit** [reply]
That's really great, because I am a Chinese user, do you have any plans for the Chinese document?

**Reply by B4nan** [reply]
Here you go 0aaaa4fe7087e874cdb97b81ddf6c9da90def259

**Reply by 0x0bit** [reply]
Thank you very much. I tried this new feature today, but it seems to have a problem.
this is my code

Entity
```js
@Filter({
  name: 'hasDeleted',
  cond: { deletedAt: { $eq: null } },
  default: true,
})
export class OrganizationEntity extends BaseEntity {
  [EntityRepositoryType]?: OrganizationRepository;

  @Property({ comment: '组织logo', nullable: true })
  logo: string = '';
……
}
```

```js
async findAll(query: QueryOrganizationDto): Promise<IHttpResultPage<OrganizationDTO>> {
    const { province, city, district, street, limit, offset, ...where } = query;
    const qb = this.orgRepository.createQueryBuilder('o').select('o.*').innerJoin('o.region', 'r');
    await qb.applyFilters();
    qb.andWhere(where);
    if (province) qb.andWhere({ 'r.province': province });
    if (city) qb.andWhere({ 'r.city': city });
    if (district) qb.andWhere({ 'r.district': district });
    if (street) qb.andWhere({ 'r.street': street });
    const res = await qb.clone().count('id', true).execute('get');
    const total = res.count;

    if ('limit' in query) qb.limit(limit);
    if ('skip' in query) qb.offset(offset);

    const items = await qb.getResult();
    return { items, total };
  }
```


<img width="1064" alt="image" src="https://github.com/user-attachments/assets/a9d5198b-c961-421e-9642-b79981d19870">

This seems normal, but no result is returned, and my api excuse will always be in LOADING state.
When I execute `console.log(await qb.applyFilters())`, The printed result is the execution result

<img width="1222" alt="image" src="https://github.com/user-attachments/assets/e1c8db6d-76b2-4974-9b1c-2115a913e6b2">

**Reply by B4nan** [reply]
Oh right, the method is async so it cannot return the QB without executing it, since the QB is awaitable (which is subject to change btw #4876). 

Fixed here by not returning anything from `qb.applyFilters` via 6be2603b27eb861bdbc90a1c29000a160d523aaf

**Reply by 0x0bit** [reply]
When will the new version be released after the issue is fixed?

**Reply by B4nan** [reply]
You can use the dev version, every commit to master is published.

**Reply by 0x0bit** [reply]
thanks
